src/writer_postgres.py

```
@@ -271,6 +274,15 @@ def write(topic_name: str, message: Dict[str, Any]) -> Tuple[bool, Optional[str]
             _logger.debug("psycopg2 not available - skipping actual Postgres write")
             return True, None
 
+        # TRACE-level payload logging (only when we intend to write)
+        if _logger.isEnabledFor(TRACE_LEVEL):
+            try:
+                _logger.trace(  # type: ignore[attr-defined]
+                    "Postgres payload topic=%s payload=%s", topic_name, json.dumps(message, separators=(",", ":"))
+                )
+            except Exception:  # pragma: no cover - defensive
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)